project/Teacher.cs

```diff
@@ -17,21 +17,52 @@ public class Teacher
         public string Name { get; set; }
         private static int nextId = 1;
         public int TeacherID { get; private set; }
-        public List<Subject> Subject { get; set; }
+        private List<Subject> _subjects = new();
+        public List<Subject> Subjects
+        {
+            get => new(_subjects); // Encapsulation: Read-only access
+            private set => _subjects = value;
+        }
+
         public Availability AvailabilityStatus { get; set; }
 
-        public Teacher(List<Subject> subjects, string name)
+        public Teacher(string name)
         {
-            Name = name;
+            Name = name ?? throw new ArgumentException("Name cannot be null or empty.");
```

Comment:
pointless, since name from the parameter is already non-nullable

Reply:
Okay . Resolved